packages/lit-html/src/async-directive.ts

```diff
@@ -303,6 +302,7 @@ export abstract class AsyncDirective extends Directive {
     parent: Disconnectable,
     attributeIndex: number | undefined
   ) {
+    this.isConnected = (part.options as PrivateRenderOptions).isConnected;
```

Comment:
Should we consider naming the private key in renderOptions more obtusely since users can pass custom values here and we don't want to conflict?

Reply:
I think it's basically fine to conflict if we always overwrite. It's the same as ignore the user-provided value. If we only sometimes overwrote it'd be an issue, IMO.

Reply:
Since this is an implementation detail that we don't want anyone to even be reading from, I went ahead and renamed it to `_$isConnected` and made it a `stableProperty`.